### PR TITLE
Disable disk watermarks on REST tests

### PR DESCRIPTION
**Pull request by ywelsch**
REST tests use the default OOTB low/high disk watermarks of 85%/90%, which can make some tests fail if run on a machine with a fuller disk. This PR changes the watermarks in the same way as in IntegTestCase so that they're essentially ignored.

### CODE REVIEW

**Review by jasontedor — CHANGES_REQUESTED** [review]
There was a workaround added for a specific test in 4bc8feea1dc0067c334798f900587900ae8867d7. Can you remove that?

**Comment by ywelsch** [comment]
800px-us_navy_110514-n-wl435-347_chief_navy_diver_billy_goold_gives_the_ok_to_signify_his_equipment_is_working_properly_before_fully_submerging_himself_u

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.shrink/10_basic.yaml [hunk 1]
```diff
@@ -1,26 +1,8 @@
...
-    - do:
-        cluster.put_settings:
-          body:
-            persistent:
-              cluster.routing.allocation.disk.threshold_enabled: null
-          flat_settings: true
-
----
 "Shrink index via API":
+  - skip:
```
Comment (jasontedor) [inline rest-api-spec/src/main/resources/rest-api-spec/test/indices.shrink/10_basic.yaml]: I don't think we need the skip now though? I'm assuming that you just picked that up in the revert.

**Review by jasontedor — CHANGES_REQUESTED** [review]
Almost there. :smile:

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by ywelsch** [comment]
Thanks @jasontedor

**Pull request merged**